aopy/preproc/base.py

```diff
@@ -331,12 +332,15 @@ def trial_align_data(data, trigger_times, time_before, time_after, samplerate):
         if np.isnan(t0):
             continue
         # sub = subvec(data, t0, n_samples, samplerate)
-        trial_data = np.empty((n_samples,data.shape[1]))
+        trial_data = np.zeros((n_samples,data.shape[1]))*np.nan
         idx_start = int(np.floor(t0*samplerate))
         idx_end = min(data.shape[0], idx_start+n_samples)
-        trial_data[:idx_end-idx_start,:] = data[idx_start:idx_end,:]
+        if idx_start < 0:
+            trial_data[-idx_start:idx_end-idx_start] = data[:idx_end,:]
```

Comment:
will data always have idx_end data points?

Reply:
yes because of `min(data.shape[0], ...)` a couple lines above